packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -393,45 +397,72 @@ class _DefaultPub implements Pub {
     String? flutterRootOverride,
   }) async {
     int exitCode;
+    if (printProgress) {
+      _logger.printStatus('Running "flutter pub $command" in ${_fileSystem.path.basename(directory)}...');
+    }
 
-    _logger.printStatus('Running "flutter pub $command" in ${_fileSystem.path.basename(directory)}...');
     final List<String> pubCommand = _pubCommand(arguments);
     final Map<String, String> pubEnvironment = await _createPubEnvironment(context, flutterRootOverride);
-    try {
-      final io.Process process;
-      final io.Stdio? stdio = _stdio;
-
-      if (stdio != null) {
-        // Omit mode parameter and direct pub output to [Pub._stdio] for tests.
-        process = await _processUtils.start(
-          pubCommand,
-          workingDirectory: _fileSystem.path.current,
-          environment: pubEnvironment,
-        );
-
-        final StreamSubscription<List<int>> stdoutSubscription =
-          process.stdout.listen(stdio.stdout.add);
-        final StreamSubscription<List<int>> stderrSubscription =
-          process.stderr.listen(stdio.stderr.add);
-
-        await Future.wait<void>(<Future<void>>[
-          stdoutSubscription.asFuture<void>(),
-          stderrSubscription.asFuture<void>(),
-        ]);
+    if (!retry) {
```

Comment:
Are we sure we need this?

If I'm correct retry is only set for the `--offline` mode, and pub itself will never retry enything in offline mode (as it won't make any requests)

Reply:
Good catch, yea, I was not sure about this. I will remove it.